http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -166,10 +168,14 @@ protected String getRedirectUri(String state) {
 
             List<String> forwardableQueryParams = Arrays.asList(LOGIN_HINT, DOMAIN_HINT, KC_IDP_HINT, PROMPT, MAX_AGE, UI_LOCALES, SCOPE);
             List<NameValuePair> forwardedQueryParams = new ArrayList<>(forwardableQueryParams.size());
+            Set<String> allScopes = new HashSet<>();
+            if (deployment.getScope() != null && !deployment.getScope().isEmpty()){
+                allScopes.addAll(Arrays.asList(deployment.getScope().split("\\s+")));
+            }
             for (String paramName : forwardableQueryParams) {
                 String paramValue = getQueryParamValue(facade, paramName);
                 if (SCOPE.equals(paramName)) {
-                    paramValue = addOidcScopeIfNeeded(paramValue);
+                    paramValue = addDeploymentAndQueryScopes(allScopes, paramValue);
```

Comment:
So instead of `addDeploymentAndQueryScopes` here, we'd call the new helper method I mentioned above.

Reply:
I have added a function called `addScope` to add a given string of scopes to a Set. And since the `addDeploymentAndQueryScopes` is fixing up the order of the scopes (putting `openid` in the beginning) and adding all scopes to the same string, I will rename it to `combineAndReorderScopes`.